### ISSUE TITLE
Add a "standard" text format for AclEntry

### ISSUE THREAD

**Issue by byllyfish** [byllyfish/exacl#38]
I propose adding a new unified text format for an AclEntry 5-tuple (kind, name, perms, flags, allow).

1. The standard format parser should accept the common Linux/setfacl text form.
1. The canonical output format will be the unambiguous, unabbreviated 5-tuple.
1. Comment character is `#`.
1. The field delimiter is a `:`. The record separator is a newline.
1. Multiple permissions and flags are separated by ','.

The canonical format is `[[allow:]flags:]kind:name:perms`

allow = "allow" | "deny"
kind = "user" | "group" | "other" | "mask" | "unknown"

Flags and perms depend on the platform.

The only flag on Linux is `default`, so this format is compatible with the linux setfacl format.

Short-form format is supported by parsing abbreviations:

1. d = default
2. u = user, g = group, o = other, m = mask
3. r = read, w = write, x = execute (no commas required).

You can pass multiple entries on the command line putting them in separate quoted arguments. There is no separator for delimiting ACL entries on a single line.

----

### Linux

Linux "long text form": `<kind>:<name>:<perms>` (Reference: man acl).
```
user::rwx
user:someuser:rwx
group::rwx
group:somegrp:rwx
mask::rwx
other::rwx
```

Comments are permitted after `#`. Linux only support `allow` access.

The `setfacl` man page puts a "default:" flag on default ACL entries.

```
[d[efault]:] [u[ser]:]uid [:perms]
[d[efault]:] g[roup]:gid [:perms]
 [d[efault]:] m[ask][:] [:perms]
[d[efault]:] o[ther][:] [:perms]
```

### macOS

macOS: `<kind>:<GUID>:<name>:<id>:<allow>,<flags>:<perms>`
```
!#acl 1
group:ABCDEFAB-CDEF-ABCD-EFAB-CDEF00000059:_spotlight:89:allow:read,write,execute
user:AAAABBBB-CCCC-DDDD-EEEE-FFFF00002CF0:::deny,file_inherit,directory_inherit:read,write,execute
```

The macOS format always begins with version (!#) line. It also includes redundant fields for the user/group id since the name, id and GUID can be derived from each other. On input, you're allowed to leave some of these blank. macOS also has many more permissions and flags.

### NFSv4

NFSv4 ACL's: `<allow>:<flags>:<name>:<perms>`
```
A::user@nfsdomain.org:rxtncy
```
There is no `kind`. It's inferred from the principal name.

setfacl on FreeBSD uses the following format for NFSv4 ACL's:
```
<kind>:<name>:<perms>:<flags>:<allow>
```

`kind` includes `owner@`, `group@`, and `everyone@`. When using these, the `name` field is omitted.

**Comment by byllyfish** [comment]
If there's a ':' in a name, it should be rendered using the octal escape. Non-printable ASCII characters should also be rendered using octal escapes.

The current code returns "@tag:N" name for unknown tag values. This needs to be cleaned up.

**Comment by byllyfish** [comment]
Examples of allowed inputs and their canonical forms:

`u:admin:rwx` => `allow::user:admin:read,write,execute`
`g::rw` => `allow::group::read,write`
`default:user:admin:r` => `allow:default:user:admin:read`
`d:group:admin:w` => `allow:default:group:admin:write`
`deny::u:self:x` => `deny::user:self:execute`

If you start with `allow` or `deny` (which are never abbreviated), you must include the empty flags field. For example, the following is NOT accepted:

`deny:u:admin:rwx` => Error: Missing flags field.

If user testing says that this is acceptable, I can relax this later.

**Comment by byllyfish** [comment]
Implemented PR #41
